src/server/index.js

```diff
@@ -22,6 +22,9 @@ if (!process.env.NEXTAUTH_URL) {
 }
 
 export default async (req, res, userSuppliedOptions) => {
+  if (userSuppliedOptions.logger) {
+    logger.setLogger(userSuppliedOptions.logger)
+  }
```

Comment:
Providing this in `userSuppliedOptions` would make this totally backwards compatible and opt-in.